tests/phpunit/tests/rest-api/rest-attachments-controller.php

```diff
@@ -2262,4 +2262,45 @@ static function () {
 			}
 		);
 	}
+
+	/**
+	 * @ticket 55443
+	 */
+	public function test_image_sources_to_rest_response() {
+
+		$attachment_id = self::factory()->attachment->create_upload_object( $this->test_file );
+		$metadata      = wp_get_attachment_metadata( $attachment_id );
+		$request       = new WP_REST_Request();
+		$request['id'] = $attachment_id;
+		$controller    = new WP_REST_Attachments_Controller( 'attachment' );
+		$response      = $controller->get_item( $request );
+
+		$this->assertInstanceOf( 'WP_REST_Response', $response );
+
+		$data       = $response->get_data();
+		$mime_types = array(
+			'image/jpeg',
+			'image/webp',
+		);
+
+		foreach ( $data['media_details']['sizes'] as $size_name => $properties ) {
+			if ( ! isset( $metadata['sizes'][ $size_name ]['sources'] ) ) {
+				continue;
+			}
+
+			$this->assertArrayHasKey( 'sources', $properties );
+			$this->assertIsArray( $properties['sources'] );
+
+			// WebP may not be supported by the server, in which case it will be stripped from the results.
+			if ( wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+				foreach ( $mime_types as $mime_type ) {
```

Comment:
Is this entire block only relevant when webp is supported or would this only apply when `$mime_type === 'image/webp'`?  Are these assertions not also valuable for `image/jpeg`?  

Ideally I think we could remove the conditional here by setting up the test in a way where we control the support in a predictable way E.g. `it_should_allow_the_upload_of_a_web_p_image_if_at_least_one_editor_supports_the_format`

Then maybe we would have two separate test cases: one for where it is supported and one where it is not 🙂

Reply:
I have adjusted the WebP support conditions so if the server does not support WebP, then the test only tests the `image/jpeg` mime type.

Does it make sense, or do I need to add two separate test cases?